acceptance/tests/api-gateway_v2/api_gateway_v2_test.go

```diff
@@ -0,0 +1,186 @@
+				"global.experiments[0]":        "resource-apis",
+			}
+
+			releaseName := helpers.RandomName()
+			consulCluster := consul.NewHelmCluster(t, helmValues, ctx, cfg, releaseName)
+
+			consulCluster.Create(t)
+
+			// Override the default proxy config settings for this test
+			consulClient, _ := consulCluster.SetupConsulClient(t, c.secure)
+			_, _, err := consulClient.ConfigEntries().Set(&api.ProxyConfigEntry{
+				Kind: api.ProxyDefaults,
+				Name: api.ProxyConfigGlobal,
+				Config: map[string]interface{}{
+					"protocol": "http",
+				},
+			}, nil)
+			require.NoError(t, err)
+
+			logger.Log(t, "creating api-gateway resources")
+			out, err := k8s.RunKubectlAndGetOutputE(t, ctx.KubectlOptions(t), "apply", "-k", "../fixtures/bases/api-gateway-v2")
+			require.NoError(t, err, out)
+			helpers.Cleanup(t, cfg.NoCleanupOnFailure, cfg.NoCleanup, func() {
+				// Ignore errors here because if the test ran as expected
+				// the custom resources will have been deleted.
+				k8s.RunKubectlAndGetOutputE(t, ctx.KubectlOptions(t), "delete", "-k", "../fixtures/bases/api-gateway-v2")
+			})
+
+			// Create certificate secret, we do this separately since
+			// applying the secret will make an invalid certificate that breaks other tests
+			logger.Log(t, "creating certificate secret")
+			out, err = k8s.RunKubectlAndGetOutputE(t, ctx.KubectlOptions(t), "apply", "-f", "../fixtures/bases/api-gateway-v2/certificate.yaml")
+			require.NoError(t, err, out)
+			helpers.Cleanup(t, cfg.NoCleanupOnFailure, cfg.NoCleanup, func() {
+				// Ignore errors here because if the test ran as expected
+				// the custom resources will have been deleted.
+				k8s.RunKubectlAndGetOutputE(t, ctx.KubectlOptions(t), "delete", "-f", "../fixtures/bases/api-gateway-v2/certificate.yaml")
+			})
+
+			// patch certificate with data
+			logger.Log(t, "patching certificate secret with generated data")
+			certificate := generateCertificate(t, nil, "gateway.test.local")
+			k8s.RunKubectl(t, ctx.KubectlOptions(t), "patch", "secret", "certificate", "-p", fmt.Sprintf(`{"data":{"tls.crt":"%s","tls.key":"%s"}}`, base64.StdEncoding.EncodeToString(certificate.CertPEM), base64.StdEncoding.EncodeToString(certificate.PrivateKeyPEM)), "--type=merge")
+
+			// We use the static-client pod so that we can make calls to the api gateway
+			// via kubectl exec without needing a route into the cluster from the test machine.
+			logger.Log(t, "creating static-client pod")
+			k8s.DeployKustomize(t, ctx.KubectlOptions(t), cfg.NoCleanupOnFailure, cfg.NoCleanup, cfg.DebugDirectory, "../fixtures/bases/static-client")
+
+			logger.Log(t, "creating target tcp server")
+			k8s.DeployKustomize(t, ctx.KubectlOptions(t), cfg.NoCleanupOnFailure, cfg.NoCleanup, cfg.DebugDirectory, "../fixtures/bases/static-server-tcp")
+
+			logger.Log(t, "creating tcp-route")
+			k8s.RunKubectl(t, ctx.KubectlOptions(t), "apply", "-f", "../fixtures/cases/api-gateways/tcproute/route.yaml")
```

Comment:
I think this route needs to be `api-gateways-v2` right? because this will end up using the kubesig route?